tests/framework/installer/ceph_manifests.go

```diff
@@ -565,6 +572,921 @@ subjects:
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: 'psp:rook'
+subjects:
+  - kind: ServiceAccount
+    name: rook-csi-rbd-provisioner-sa
+    namespace: ` + namespace + `
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: rook-csi-rbd-plugin-sa-psp
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: 'psp:rook'
+subjects:
+  - kind: ServiceAccount
+    name: rook-csi-rbd-plugin-sa
+    namespace: ` + namespace + `
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: rook-csi-cephfs-provisioner-sa-psp
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: 'psp:rook'
+subjects:
+  - kind: ServiceAccount
+    name: rook-csi-cephfs-provisioner-sa
+    namespace: ` + namespace + `
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: rook-csi-cephfs-plugin-sa-psp
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: 'psp:rook'
+subjects:
+  - kind: ServiceAccount
+    name: rook-csi-cephfs-plugin-sa
+    namespace: ` + namespace + `
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: rook-ceph-operator
+  namespace: ` + namespace + `
+  labels:
+    operator: rook
+    storage-backend: ceph
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: rook-ceph-operator
+  template:
+    metadata:
+      labels:
+        app: rook-ceph-operator
+    spec:
+      serviceAccountName: rook-ceph-system
+      containers:
+      - name: rook-ceph-operator
+        image: rook/ceph:` + m.imageTag + `
+        args: ["ceph", "operator"]
+        env:
+        - name: ROOK_LOG_LEVEL
+          value: INFO
+        - name: ROOK_CEPH_STATUS_CHECK_INTERVAL
+          value: "5s"
+        - name: ROOK_MON_HEALTHCHECK_INTERVAL
+          value: "10s"
+        - name: ROOK_MON_OUT_TIMEOUT
+          value: "15s"
+        - name: NODE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        # CSI enablement
+        - name: ROOK_CSI_ENABLE_CEPHFS
+          value: "true"
+        - name: ROOK_CSI_ENABLE_RBD
+          value: "true"
+        volumeMounts:
+        - mountPath: /etc/ceph-csi/rbd
+          name: csi-rbd-config
+        - mountPath: /etc/ceph-csi/cephfs
+          name: csi-cephfs-config
+      volumes:
+      - name: csi-rbd-config
+        configMap:
+          name: csi-rbd-config
+      - name: csi-cephfs-config
+        configMap:
+          name: csi-cephfs-config
+`
+}
+
+// GetRookOperatorForOpenshift returns rook Operator manifest for openshift
+func getRookOperatorForOpenshiftYaml(namespace string, m *CephManifestsMaster) string {
```

Comment:
@travisn I remember from you comment in my fork [here](https://github.com/petr-balogh/rook/commit/163216fedc75c90a1c4b74e68869f377403f008a#r34344737) that you would like to have separate function and have the yaml defined there for openshift for better readability than compose this yaml because of some parts are different.

In this approach I see a lot of duplicity and it's probably not good for maintenance as ppl will need to add for both places.  Do you have in mind different solution you prefer here?

Reply:
Agreed we don't want to duplicate all of this. The main differences are the two SCCs and an env var in the rook-ceph-operator deployment. Sounds great if we can just add conditionals to add those for openshift.

Reply:
@travisn I changed the logic here to do not duplicate the code, is it OK like this?